From 2fee2f1bf1f4bd009c95792f28308ce8527f69ff Mon Sep 17 00:00:00 2001
From: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Date: Wed, 8 Jan 2025 16:48:11 +0200
Subject: [PATCH 2/2] net/netmon: Allow excluding custom interfaces

Providing a regex env variable ensures that any other specific
interface that interferes with tailscale can be disabled on startup.

Updates: #1552

Signed-off-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
---
 net/netmon/state.go | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/net/netmon/state.go b/net/netmon/state.go
index 4f8788907..4b47a7190 100644
--- a/net/netmon/state.go
+++ b/net/netmon/state.go
@@ -9,6 +9,8 @@
 	"net"
 	"net/http"
 	"net/netip"
+	"os"
+	"regexp"
 	"runtime"
 	"slices"
 	"sort"
@@ -36,6 +38,13 @@ func isLoopback(nif *net.Interface) bool { return nif.Flags&net.FlagLoopback !=
 
 func isProblematicInterface(nif *net.Interface) bool {
 	name := nif.Name
+
+	if regex := os.Getenv("TS_NETMON_IGNORE"); regex != "" {
+		if match, _ := regexp.MatchString(regex, name); match {
+			return true
+		}
+	}
+
 	// Don't try to send disco/etc packets over zerotier; they effectively
 	// DoS each other by doing traffic amplification, both of them
 	// preferring/trying to use each other for transport. See:
-- 
2.51.0

